block-storage/src/main/scala/coop/rchain/blockstorage/BlockStore.scala

```diff
@@ -1,31 +1,39 @@
 package coop.rchain.blockstorage
 
-import cats.Applicative
+import cats.effect.Sync
 import cats.syntax.all._
-import coop.rchain.casper.protocol.{ApprovedBlock, BlockMessage}
+import coop.rchain.blockstorage.dag.codecs.{codecBlockHash, codecBlockMessage}
+import coop.rchain.casper.protocol.{BlockMessage, BlockMessageProto}
 import coop.rchain.models.BlockHash.BlockHash
+import coop.rchain.shared.Compression
+import coop.rchain.shared.syntax._
+import coop.rchain.store.{KeyValueStoreManager, KeyValueTypedStore}
+import net.jpountz.lz4.{LZ4CompressorWithLength, LZ4DecompressorWithLength}
 
-trait BlockStore[F[_]] {
-  def get(blockHash: BlockHash): F[Option[BlockMessage]]
-
-  def put(f: => (BlockHash, BlockMessage)): F[Unit]
+object BlockStore {
+  type BlockStore[F[_]] = KeyValueTypedStore[F, BlockHash, BlockMessage]
```

Comment:
Although it makes sense to use BlockStore explicitly, for now better to minimize changes. I suggest this definition which will enable us to use BlockStore implicitly as now.

Object name start with lowercase so it's not shadowing `BlockStore[F]` constructor for implicit use as we have now.
```scala
object blockStore {
  type BlockStore[F[_]] = KeyValueTypedStore[F, BlockHash, BlockMessage]

  // Simulates usual apply definition to get the instance `BlockStore[F]` from implicit value
  def BlockStore[F[_]](implicit instance: BlockStore[F]): instance.type = instance
  ...
}
```